server/game/cards/locations/04/vaestolorro.js

```diff
@@ -0,0 +1,35 @@
+const DrawCard = require('../../../drawcard.js');
+
+class VaesTolorro extends DrawCard {
+    setupCardAbilities(ability) {
+        this.interrupt({
+            when: {
+                onCharacterKilled: (event, player, card) => {
+                    if(!card.getPower() >= 1) {
+                        return false;
+                    }
+                    this.pendingCard = card;
+                    
+                    return true;
+                }
+            },
+            cost: ability.costs.kneelSelf(),
+            handler: () => {
+                if(this.pendingCard.getPower() >= 2) {
+                    var power = this.pendingCard.getStrength() === 0 ? 2 : 1;
+                } else {
+                    power = 1;
```

Comment:
Not sure how @cryogen feels style-wise but I'm generally not fond of this construct when a variable is defined in one block and then used in another sibling-level block. It's technically valid Javascript but doesn't read well. I'd suggest either declare the variable above the `if` block with initial value 1 (which gets rid of the need for the `else`) or using `Math.min` to one-line it.